nucypher/network/nodes.py

```diff
@@ -348,8 +348,12 @@ def read_nodes_from_storage(self) -> List:
         stored_nodes = self.node_storage.all(federated_only=self.federated_only)  # TODO: #466
 
         restored_from_disk = []
-
         for node in stored_nodes:
+            node_domain = node.domain.decode('utf-8')
+            if node_domain != self.learning_domain.encode():
+                self.log.warn(f"Restored {node} metadata from storage, but appears to be for domain '{node_domain}'; "
```

Comment:
Seems like solid logic.  Though it occurs to me that result is likely to either be no warnings at all, or a ton of warnings.  Maybe it makes sense to build a list of the nodes and emit a single warning?

Reply:
That's a very good idea. Another interesting questions is what to do about it. Should we "clean" the storage or do nothing?

Reply:
✅